custom_components/micro_air_easytouch/climate.py

```
@@ -248,35 +257,54 @@ async def async_set_temperature(self, **kwargs: Any) -> None:
         """Set new target temperature."""
         ble_device = async_ble_device_from_address(self.hass, self._mac_address)
         if not ble_device:
-            _LOGGER.error("Could not find BLE device")
-            return
+            raise HomeAssistantError("Could not find BLE device")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)